keps/sig-instrumentation/0034-distributed-tracing-kep.md

```diff
@@ -81,119 +92,176 @@ Distributed tracing provides a single window into latency information from acros
 
 * Replace existing logging, metrics, or the events API
 * Trace operations from all Kubernetes resource types in a generic manner (i.e. without manual instrumentation)
+* Change metrics or logging (e.g. to support trace-metric correlation)
+* Access control to tracing backends
 
 ## Proposal
 
 ### Architecture
 
 #### Tracing API Requests
 
-In the traditional tracing model, a client sends a request to a server and receives a response back.  Even though Kubernetes "controllers" don't follow this model (more on that later), the kube-apierver and backing storage (e.g. etcd3) do.  To enable traces to be collected for API requests, the following must be true:
+In the traditional tracing model, a client sends a request to a server and receives a response back.  The kube-apierver and backing storage (e.g. etcd3) follow this model, so we can easily add tracing for API Server requests.  To enable traces to be collected for API requests, the following must be true:
 
 1. The apiserver must propagate the http context of incoming requests through its function stack to the backing storage
 1. Kubernetes client libraries must allow passing a context with API requests
```

Comment:
This will be dramatically misleading if webhooks don't propagate. So, apiserver needs to propagate the context to webhooks, and they need to propagate that context to any calls they make (e.g., re-entrant calls to apiserver).

Additionally, the aggregator needs to propagate the context to aggregated apiservers.

I agree with this use of tracing, which is super useful and theoretically sound.

Reply:
I added a note that webhooks should propagate context for reentrant calls.

I made the language around propagation more general to say that the API Server should propagate context to all clients.